tree/dataframe/src/RDFColumnRegister.cxx

```diff
@@ -62,11 +113,14 @@ bool RColumnRegister::IsDefineOrAlias(std::string_view name) const
 ////////////////////////////////////////////////////////////////////////////
 /// \brief Add a new defined column.
 /// Internally it recreates the map with the new column, and swaps it with the old one.
-void RColumnRegister::AddDefine(std::shared_ptr<RDFDetail::RDefineBase> column)
+void RColumnRegister::AddDefine(std::shared_ptr<RDFDetail::RDefineBase> define)
 {
    auto newDefines = std::make_shared<DefinesMap_t>(*fDefines);
```

Comment:
it's probably still too terse, but the explanation is in the docs of the data member:

```cpp
   /// Immutable collection of Defines, can be shared among several nodes.
   /// The pointee changes if a new Define node is added to the RColumnRegister.
std::shared_ptr<DefinesMap_t> fDefines;
```

It's not just about `fDefines` though, this is in general how `RColumnRegister` manages its data members. I'll add the following explanation in the class docs (in a different PR, if that's ok):

> Several components of an RDF computation graph make use of a column register. It keeps track of which columns have been defined, varied or aliased at each point of the computation graph.
>
> In many cases, the contents of the different column register instances are the same or only differ by a single extra defined/varied/aliased column. For this reason, in order to avoid unnecessary data duplication, `fDefines`, `fAliases`, `fVariations` and `fColumnNames` are all `shared_ptr<const T>` that (whenever possible) are shared across `RColumnRegister` instances that are part of the same computation graph. If a new column, alias or variation is added between one node and the next, then the new node contains a new instance of a `RColumnRegister` that shares all data members with the previous instance except for the one data member that needed updating, which is replaced with a new updated instance.

Reply:
(this discussion can continue at https://github.com/root-project/root/pull/11297 , marking as resolved)